app/controllers/application_controller.rb

```diff
@@ -5,6 +5,178 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "secret"
   end
 
-end
\ No newline at end of file
+get '/' do
+erb :homepage
+  #homepage
+end
+
+get '/signup' do
+   if logged_in?
+     redirect '/tweets'
+   else
+  erb :"/users/signup"
+end
+end
+
+get '/login' do
+  if logged_in?
+    redirect '/tweets'
+  else
+  erb :"/users/login"
+end
+end
+
+get '/tweets' do
+  if logged_in?
+  @user = User.find(session[:user_id])
+ session[:user_id] = @user.id
+  erb :"/tweets/tweets"
+else
+  redirect '/users/login'
+end
+end
+
+ get '/tweets/new' do
+  #  raise session.inspect
+  # @user = User.find(session[:user_id])
+   if   logged_in?
+     erb :"/tweets/new"
+   else
+     redirect '/login'
+   end
+ end
+
+ get '/tweets/:id' do
+
+    @tweet = Tweet.find(params[:id])
+    # if logged_in?
+   if session[:user_id] == @tweet.user_id
+
+     @user = User.find(session[:user_id])
+   erb :"tweets/show"
+ elsif logged_in? && session[:user_id] != @tweet.user_id
+   @user = User.find(session[:user_id])
+   erb :"tweets/show"
+
+ else
+
+   redirect '/login'
+ end
+ end
+
+ get '/tweets/:id/edit' do
+   if logged_in?
+     @user = User.find(session[:user_id])
+     @tweet = Tweet.find(params[:id])
+     if @user.id == @tweet.user_id
+
+     erb :"/tweets/edit"
+   else
+     redirect '/tweets'
+   end
+ else
+   redirect '/login'
+ end
+ end
+
+ get "/users/:slug" do
+   @user = User.find_by_slug(params[:slug])
+     erb :'/users/show'
+ end
+
+get '/logout' do
+    session.clear
+  redirect '/login'
+ end
+
+patch '/tweets/:id' do
+  # binding.pry
+  if logged_in?
+   @tweet=Tweet.find(params[:id])
+     if !params[:content].empty?
```

Comment:
I would contend that you still want to double check the user's association to the tweet here. There are other ways to submit a PATCH/POST request than going through an HTML form...I would